### PR TITLE
Proposed Feature Edit Character List

### PR DESCRIPTION
**Pull request by zachlankton**
This PR/Branch is a demo of a feature to add an editable character list to the teachers page.

This will allow the teacher to create their own character lists that will still randomly be assigned as usual.

Additionally, logic has been provided for if the teacher empties the list, the character name will default to the users real name.

feat-edit-char-list

### CODE REVIEW

**Review by mssiegel — APPROVED** [review]
This is incredible!!! I love the design. AMAZING!!!!! THIS FEATURE IS EXACTLY WHAT WE NEED!!!!!!  I left some minor comments, feel free to address them.    After you can go ahead and merge or ask for another review prior to merging. Whichever you prefer.

File: src/utils/classrooms.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ export function getClassroom(classroom: string) {
 }
 
 export function getRandom<T>(arr: Array<T>): T {
+  if (arr.length === 0) return null;
   return arr[Math.floor(Math.random() * arr.length)];
 }
```
[the remark this answers is not in the document]
Comment (zachlankton) [reply src/utils/classrooms.tsx:20]: You are correct, this fixed my problem initially when the character list was empty, but the real problem was in the student pairing logic.  The loop that assigns random character names will loop forever if there are not atleast 2 characters.  I will add the check there to fix this.

File: src/components/pages/TeachersPage/UnpairedStudentsList.tsx [hunk 2]
```diff
@@ -88,37 +104,73 @@ export default function UnpairedStudentsList({ socket }) {
...
+            Pair up students
+          </Button>
+        </Box>
+        <Box sx={{ display: 'flex', justifyContent: 'center' }}>
+          <Button
+            variant='contained'
+            size='large'
+            color='success'
+            sx={{ mt: 2 }}
+            startIcon={<ChatIcon />}
```
[the remark this answers is not in the document]
Comment (zachlankton) [reply src/components/pages/TeachersPage/UnpairedStudentsList.tsx:167]: Sounds good, will do!

File: src/components/pages/TeachersPage/UnpairedStudentsList.tsx [hunk 3]
```diff
@@ -25,6 +28,19 @@ const CHARACTERS = [
 
 export default function UnpairedStudentsList({ socket }) {
   const [unpairedStudents, setUnpairedStudents] = useState<Student[]>([]);
+  const [characters, setCharacters] = useState(CHARACTERS);
+  const [openCharacterModal, setOpenCharacterModal] = useState(false);
+  const handleCloseCharacterModal = () => setOpenCharacterModal(false);
+  const charListRef = useRef<HTMLTextAreaElement>();
+
+  const setCharList = () => {
+    if (!charListRef.current) return;
```
[the remark this answers is not in the document]
Comment (zachlankton) [reply src/components/pages/TeachersPage/UnpairedStudentsList.tsx:37]: It's probably not necessary.  This happens to me in my ERP code base, Typescript complains that `current` might be null and needs to be checked at run time.  Force of habit. I can take it out.

**Review by mssiegel — APPROVED** [review]
Looks great to me! Let's merge!

**Pull request merged**